dashboard/app/controllers/registrations_controller.rb

```diff
@@ -137,4 +141,28 @@ def update_params(params)
       races: []
     )
   end
+
+  # resource_name, resource, devise_mapping, and resource_class are defined here mostly following
+  # https://pupeno.com/2016/04/26/show-a-devise-log-in-or-sign-up-forms-in-another-page/
+  # in order to allow the upgrade action to render an error page correctly
```

Comment:
Just tried this now, and `prepend_before_action :authenticate_scope!, only: [:edit, :update, :destroy, :upgrade]` seems to work fine, so I'm going with that. Good suggestion!